driven/storage/adapter.go

```diff
@@ -310,18 +355,30 @@ func (sa Adapter) addTokenToUserWithContext(ctx context.Context, orgID string, a
 	return nil
 }
 
-func (sa Adapter) removeTokenFromUserWithContext(ctx context.Context, orgID string, appID string, token string, userID string) error {
+func (sa Adapter) removeTokenFromUserWithContext(ctx context.Context, orgID string, appID string, token string, userID string, tokenType *string) error {
 	filter := bson.D{
 		primitive.E{Key: "org_id", Value: orgID},
 		primitive.E{Key: "app_id", Value: appID},
 		primitive.E{Key: "user_id", Value: userID},
 	}
 
-	update := bson.D{
-		primitive.E{Key: "$set", Value: bson.D{
-			primitive.E{Key: "date_updated", Value: time.Now().UTC()},
-		}},
-		primitive.E{Key: "$pull", Value: bson.D{primitive.E{Key: "firebase_tokens", Value: bson.D{primitive.E{Key: "token", Value: token}}}}},
+	update := bson.D{}
+	if tokenType != nil {
```

Comment:
The `tokenType` passed into this function should prepend the `"_tokens"` key in the update object.